users/dierkes/preprocessing/segment_cutting.py

```diff
@@ -0,0 +1,107 @@
+import pathlib
+import soundfile as sf
+import numpy as np
+import json
+import multiprocessing
+import argparse
+import tqdm
+import os
+import shutil
+import logging
+
+from sisyphus import Job, Task, tk
+
+class CutSpeechSegmentsFromAudio(Job):
+    """
+    This Job uses a transcription file to cut all segments containing speech from the audio files
+    and stitches them together to files of a target length. Segments that are
+    longer than the target_length are stored as individual files.
+
+
+    """
+    def __init__(self, audio_dir, segments_dir, target_length=20, n_workers=32, file_extension='.wav'):
+        """
+
+        :param audio_dir: path to all the audio files. Subfolders are not yet supported in audio
+            directory
+        :param segments_dir: path to the transcriptions. Subfolders are not yet supported in
+            transcription directory
+        :param target_length: target length of the stitched extracted speech
+        :param n_workers:
+        :param file_extension: currently supported are wav, flac and mp3
+        """
+        self.audio_dir = audio_dir
+        self.segments_dir = segments_dir
+        self.target_length = target_length
+        self.n_workers = n_workers
+        self.file_extension = file_extension
+        assert file_extension in ['.wav', '.flac', '.mp3']
+
+        self.out_audio_folder = self.output_path("audio/", directory=True)
+        self.cut_rqmt = {'time': 8, 'mem': 8, 'cpu': 4}
```

Comment:
I am confused about this: If I schedule jobs with more than 4 cpu it tells me that those requirements cannot be met by the cluster. What does cpu mean here? _Actual_  cpu's or number of cores of a single cpu? I am using multiple processes in my code, so one cpu with multiple cores would already be sufficient to achieve parallelism.

Reply:
In that case you have wrong Sisyphus settings with respect to the SGE. Please check your `settings.py` file of your setup:
I have something like this:
```python
    return EngineSelector(engines={'short': LocalEngine(cpus=4),
                                   'long': SonOfGridEngine(default_rqmt={'cpu' : 1, 'mem' : 2, 'gpu' : 0, 'time' : 1})},
                                  default_engine='long')
```
so if for any reason you run the job on "short", it will not have more than 4 cpus. But the SGE is usually unlimited.

Reply:
Regarding the number of cores per CPU, I'm not sure what a reasonabe assumption would be. Do you know @JackTemaki?

Reply:
I figured out the problem with the requirements not being met and fixed it

Reply:
@JackTemaki do you think this is alright? If so, I'd merge the PR.